### PR TITLE
Add deploy option and some other changes.

### PR DESCRIPTION
**Pull request by TV4Fun**
This adds an option to upload the generated egg to a PyPI repository
via the deploy goal. Also some rewrites to save on code reuse and adding
support for a few different distribution types. Also fixed the install command.

### CODE REVIEW

File: src/main/java/com/github/mojos/distribute/AbstractSetupCommandMojo.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    abstract void addSetupArgs(List<String> args) throws MojoExecutionException;
+
+    /* (non-Javadoc)
+     * @see org.apache.maven.plugin.AbstractMojo#execute()
+     */
+    @Override
+    public void execute() throws MojoExecutionException {
+        final File buildDirectory = Paths.get(project.getBuild().getDirectory(), "maven-python").toFile();
+        final String setupOutputCanonicalPath = project.getProperties().getProperty("python.distribute.plugin.setup.path");
```
Comment (UltimateDogg) [inline src/main/java/com/github/mojos/distribute/AbstractSetupCommandMojo.java]: can you default this to the old path (i think it was src/main/python/setup.py) for compatibility?

File: src/main/java/com/github/mojos/distribute/DeployMojo.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package com.github.mojos.distribute;
```
Comment (UltimateDogg) [inline src/main/java/com/github/mojos/distribute/DeployMojo.java:1]: can you add something to readme to describe this. Does it work with all distribution types, etc.

Comment (TV4Fun) [reply src/main/java/com/github/mojos/distribute/DeployMojo.java:1]: Done.

File: src/main/java/com/github/mojos/distribute/AbstractSetupCommandMojo.java (hunk 1 above)
Comment (TV4Fun) [reply src/main/java/com/github/mojos/distribute/AbstractSetupCommandMojo.java]: Done.

**Comment by UltimateDogg** [comment]
thanks!

**Pull request merged**